src/fsharp/TastReflect.fs

```diff
@@ -988,9 +988,10 @@ and ReflectAssembly(g, ccu: CcuThunk, location:string) as asm =
                 | None -> failwith (sprintf "TODO: didn't get back to CCU being compiled for local tcref %s" tcref.DisplayName)
             let reflAssem = ccuofTyconRef.ReflectAssembly :?> ReflectAssembly
             let tcrefR = reflAssem.TxTypeDef None tcref
-            match tinst with 
-            | [] -> tcrefR 
-            | args -> tcrefR.MakeGenericType(Array.map asm.TxTType (Array.ofList args))  
+            tcrefR
+            //match tinst with 
```

Comment:
I don't think this is right: we need to take into account the type instantiation when translating a type.  What caused you to remove the processing of ``tinst``?

Perhaps you were having trouble processing ``C<T>`` where ``C`` is instantiated at its own generic type?  If so we should detect that as a special case (we can detect that case by whether ``typeEquiv (snd (generalizeTyconRef tcref)) typ`` true)